### PR TITLE
Use a longer expiry deadline for ad-free eligibility

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
Currently, the async user credential refresh doesn't (and ideally never will) cause a page reload. As a result we sometimes serve ads to ad-free users on the initial page load. This prevents that because the cookie remains valid.

Note that the first time someone logs in with ad-free eligibility, that page will still contain ads and needs to be refreshed (or another page viewed).

## What is the value of this and can you measure success?
Ad-free eligibility is still assessed in the same way, but allowing the cookie to live longer than the user credentials default means that initial page load won't serve ads if it shouldn't.

_Update_
We're now able to control whether we strictly enforce the expiry date of the ad-free cookie, or not, via a new `ad-free-strict-expiry-enforcement` config switch.

If the switch is `OFF` (its default state), a logged-in user's initial page load will ignore the expired state of the ad-free cookie, rendering an ad-free page regardless of how long it's been since they last signed in / revalidated. 

The default subsequent re-writing of the cookie value will happen as the user's credentials are updated to reflect their current status, so if they're no longer valid for ad-free the cookie will be removed and they'll see ads in subsequent page views.

If we switch the enforcer switch `ON`, we'll serve ads on the first page view if the cookie is stale, and only rectify when we know the user is still eligible for ad-free.

## Does this affect other platforms - Amp, Apps, etc?
No, web only.

## Screenshots
N/A

## Tested in CODE?
Yes.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/user-features.js [hunk 1]
```diff
@@ -23,15 +23,19 @@ const adFreeDataIsPresent = (): boolean => {
     return !isNaN(parseInt(cookieVal, 10));
 };
 
+const timeInDaysFromNow = (daysFromNow: number): string => {
```
Comment (davidfurey) [inline static/src/javascripts/projects/commercial/modules/user-features.js]: Would `nowPlusDays` be a better name?

Comment (JustinPinner) [inline static/src/javascripts/projects/commercial/modules/user-features.js]: Well, the function returns a `getTime().toString()` and it isn't restricted to positive numbers for `daysFromNow`, although we only use positives at the moment.

Comment (davidfurey) [inline static/src/javascripts/projects/commercial/modules/user-features.js]: Fair enough.  It isn't restricted to integers either I suppose

**Review by davidfurey — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/user-features.js (hunk 1 above)
Comment (JustinPinner) [inline static/src/javascripts/projects/commercial/modules/user-features.js]: Ha. True!

**Pull request merged**